tgui/packages/tgui/interfaces/BombProcessor.js

```diff
@@ -0,0 +1,122 @@
+import { useBackend } from '../backend';
+import { Button, Flex, Icon, Modal, Section, Stack } from '../components';
+import { Window } from '../layouts';
+import { GasmixParser } from './common/GasmixParser';
+
+export const BombProcessor = (props, context) => {
+  return (
+    <Window title="Bomb Processor" width={600} height={400}>
+      <Window.Content>
+        <BombProcessorContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const BombProcessorContent = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { gasList, valve, reactionIncrement } = data;
+  const ResultantGasmix = (
+    <Section title={gasList[2].name}>
+      <Stack mh={1}>
+        <Stack.Item basis={0} grow>
+          <Section>
+            {!gasList[2].total_moles && <Modal>{'No Gas Present'}</Modal>}
+            <GasmixParser {...gasList[2]} />
+          </Section>
+        </Stack.Item>
+        <Stack.Item basis={0} grow>
+          <Stack vertical textAlign="center" fill>
+            <Stack.Item grow>
+              {reactionIncrement === 0 ? (
+                <Button
+                  height="100%"
+                  fontSize={1.25}
+                  fluid
+                  tooltip={'Valve Status: Closed.'}
+                  tooltipPosition="top"
+                  onClick={() => act('react')}
+                  disabled={!gasList[0].total_moles || !gasList[1].total_moles}>
+                  <Flex
+                    height="100%"
+                    width="100%"
+                    justify="center"
+                    direction="column">
+                    <Flex.Item>
+                      <Icon name="vial" />
+                      {'Open Valve'}
+                    </Flex.Item>
+                  </Flex>
+                </Button>
+              ) : (
+                <Button
+                  height="100%"
+                  fontSize={1.25}
+                  fluid
+                  tooltip={
+                    'Valve Status: Open. Current Reaction Increment: '
+                    + (reactionIncrement - 1)
+                  }
+                  tooltipPosition="top"
+                  onClick={() => act('react')}
+                  disabled={!gasList[0].total_moles || !gasList[1].total_moles}>
+                  <Flex
+                    height="100%"
+                    width="100%"
+                    justify="center"
+                    direction="column">
+                    <Flex.Item>
+                      <Icon name="vial" />
+                      {'React'}
+                    </Flex.Item>
+                  </Flex>
+                </Button>
+              )}
+            </Stack.Item>
+            <Stack.Item grow>
+              <Button
+                height="100%"
+                fontSize={1.25}
+                fluid
+                disabled={!valve}
+                onClick={() => act('eject')}>
+                <Flex
+                  height="100%"
```

Comment:
Want to take a try at improving tgui? Make Button's text vertically align by default when changing its height. Consult me on discord.